src/components/RangeSlider/types.ts

```diff
@@ -64,16 +65,34 @@ export type RangeSliderProps = {
   showSelectedRange?: boolean;
   showHandleLabels?: boolean;
 
-  motionBlur?: boolean;
   springOnRelease?: boolean;
+  /** Debounce interval (in ms) before calling `onDebounceChange`. */
   debounceInterval?: number;
-  axisLock?: 'x' | 'y' | '';
-  onDrag?: (val: number) => void;
+
+  /** Called immediately as slider's selection changes. */
+  onChange?: (val: number) => void;
+  /** Called `debounceInterval` ms after the most recent change of selection. */
+  onDebounceChange?: (val: number) => void;
+  /** Called when the slider's drag gesture is released */
+  onRelease?: (val: number) => void;
+
   disabled?: boolean;
   readonly?: boolean;
   min: number;
   max: number;
-  values?: number[] | ValueProp[];
+  values: number[] | ValueProp[];
   testId?: string;
   markers?: number[] | ValueProp[];
+  /**
+   * If true, the drag handle will snap to the passed-in `value`.
+   * If false, the drag handle will directly follow the mouse on drag.
+   */
+  dragHandleBehavior?: 'snapToValue' | 'followMouse';
```

Comment:
I feel like camelCase isn't right for text attributes - but maybe we could reduce it to one word either way, like
```suggestion
  dragHandleAttachment?: 'value' | 'mouse';
```